picard/script.py

```diff
@@ -175,6 +184,9 @@ class ScriptParser(object):
     _function_registry = ExtensionPoint(label='function_registry')
     _cache = {}
 
+    def __init__(self):
+        self._function_stack = []
```

Comment:
I'm not sure implementing it as a simple list is future-proof, because `list` isn't thread-safe.

There are altenatives that are thread-safe, and could prolly work:

```python
from queue impot LifoQueue

stack = LifoQueue()
stack.put('bar')
stack.put('foo')
...
if not stack.empty():
  last = stack.get()
```

Since we do `stack = stack[:-1]` but on failure where we only read last value, it could make sense to use lifo queues and be thread-safe as bonus.